File: qc-core/src/main/java/com/eager/questioncloud/core/domain/review/infrastructure/repository/QuestionReviewStatisticsRepository.java
package com.eager.questioncloud.core.domain.review.infrastructure.repository;

import com.eager.questioncloud.core.domain.review.model.QuestionReviewStatistics;

public interface QuestionReviewStatisticsRepository {
    QuestionReviewStatistics get(Long questionId);

    QuestionReviewStatistics save(QuestionReviewStatistics questionReviewStatistics);

    void deleteAllInBatch();
}

File: qc-core/src/main/kotlin/com/eager/questioncloud/core/domain/review/infrastructure/repository/QuestionReviewStatisticsRepository.kt
package com.eager.questioncloud.core.domain.review.infrastructure.repository

import com.eager.questioncloud.core.domain.review.model.QuestionReviewStatistics

interface QuestionReviewStatisticsRepository {
    fun get(questionId: Long): QuestionReviewStatistics

    fun getForUpdate(questionId: Long): QuestionReviewStatistics

    fun save(questionReviewStatistics: QuestionReviewStatistics): QuestionReviewStatistics

    fun update(questionReviewStatistics: QuestionReviewStatistics)

    fun deleteAllInBatch()
}
